### PR TITLE
Allow functions of `DistributedVariable`s to implement `custom_gradient`

### PR DESCRIPTION
**Pull request by jkyl**
Previously, variables created by functions decorated with `custom_gradient` had to pass:
```python 
isinstance(v, resource_variable_ops.ResourceVariable)
```
With this patch, they instead must pass:
```python
resource_variable_ops.is_resource_variable(v)
```
This allows functions that create [`DistributedVariable`](https://github.com/tensorflow/tensorflow/blob/a799b815e5b497fd85e36ec36e44df0f3cebf802/tensorflow/python/distribute/values.py#L429)s (which already implement the method [`_should_act_as_resource_variable`](https://github.com/tensorflow/tensorflow/blob/a799b815e5b497fd85e36ec36e44df0f3cebf802/tensorflow/python/distribute/values.py#L568)) to use `custom_gradient`, which is necessary for cross-replica batch normalization, among other things.

### CODE REVIEW

**Comment by jkyl** [comment]
> Once you've signed (or fixed any issues), please reply here (e.g. `I signed it!`) and we'll verify it.

Signed the CLA

**Comment by MarkDaoust** [comment]
@alextp, I'm not familiar with this code. Are you the right person to review this?

**Review by alextp — APPROVED** [review]

**Comment by alextp** [comment]
The change looks good, but I'd also love a unit test for it.

**Comment by jkyl** [comment]
I would be happy to submit a unit test; however since there is not yet any `custom_gradient_test.py`, it seems befitting of a separate PR.
